src/Utilities/cachingoptimizer.jl

```diff
@@ -864,15 +864,30 @@ function _get_model_attribute(
     )
 end
 
+function _throw_if_get_attribute_not_allowed(
+    model::CachingOptimizer,
+    attr;
+    needs_optimizer_map::Bool = false,
```

Comment:
Maybe don't give a default value for this keyword and set it explicitly. I feel there is less chance of bug if we set it explicitly at every call